langchain-core/src/runnables/base.ts

```diff
@@ -515,6 +523,15 @@ export abstract class Runnable<
 
     let runManager: CallbackManagerForChainRun | undefined;
     try {
+      const resolvedOrchestratorAbortBehavior =
+        options?.orchestratorAbortBehavior ??
+        OrchestratorAbortBehavior.THROW_IMMEDIATELY;
+
+      const signal =
+        resolvedOrchestratorAbortBehavior ===
+        OrchestratorAbortBehavior.THROW_IMMEDIATELY
+          ? options?.signal
+          : undefined;
```

Comment:
This special case was necessary here to avoid a breaking change to the `pipeGeneratorWithSetup` function. I _think_ it's meant to be an internal utility, but nothing marks it as internal and it's exported from a file that's included in the package exports, so I think we should regard it as public even if that wasn't the original intent.